### DISCUSSION TITLE
Basemap support for static and interactive geoshape charts

### DISCUSSION THREAD

**Discussion by gregorhd** [vega/altair#2485] [Q&A]
I saw the discussion in Vega-Lite issue [#5758](https://github.com/vega/vega-lite/issues/5758) regarding basemap support, which seems to be working even for interactive plots, but how would one implement this directly in Altair with a chart containing geoshapes?

On the static front, I got this chart to work based on a GeoPandas dataframe.

```
import geopandas as gpd
import altair as  alt
import altair_viewer
import json

gdf = gpd.GeoDataFrame(etc. etc.)

gdf_json = gdf.to_json()

json_features = json.loads(merged_json)

features = alt.Data(values=json_features['features'])

domain = ['Label1', 'Label2', 'Label3']
range_ = ['red', 'grey', 'cornflowerblue']

chart = alt.Chart(features).mark_geoshape().properties(
        projection={'type': 'mercator'},
        width=700,
        height=400
).encode(
        color=alt.Color("properties.Legend:N", scale=alt.Scale(domain=domain, range=range_), legend=alt.Legend(title='Legend', orient='top-right'))
    )

altair_viewer.display(chart)
```

Adding `.interactive(),` I faced the same problem as this user in issue [#933](https://github.com/altair-viz/altair/issues/933#issuecomment-396338850), which according to the referenced issue [#679](https://github.com/altair-viz/altair/issues/679) is due to vega-lite not, at least at that time, supporting geoshapes for interactive mode. Vega-Lite issue #5758 though would suggest that this is not current anymore, meaning one could even have an interactive geoshapes chart with a basemap?

Any pointers are much appreciated!

**Comment by jakevdp** [comment]
I don't think vega-lite has any support for scale-bound selections in geographic plots, which is what `.interactive()` attempts to add. So no, that is not supported in Altair.

**Reply by gregorhd** [reply]
Thanks, Jake! Does that mean tile basemaps for non-interactive charts are also not supported yet?

**Reply by jakevdp** [reply]
I don't know of any example of using basemaps in Altair, though for a motivated enough user it may be possible using something along the lines of the example in the linked vega issue.

**Reply by mattijn** [reply]
The example ([Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAdxoATemgAsABh1L4NMlm16lSBpjYQaALzhoQsuQ6U4katTVlk0oKzho+iBMbNhsCEFKgnAAZphRIABORiaoOgC+bkhJSAgQaADaoLJ5DuiYAB6KIHCVOEmOqhowBABMIFklZY6YAJ41dQ2OhsaY7Z0K3QjlILZsEQD6DXBQNBCuIMRIggzlAGyUAOxKTN5qfiDeOJaOuT6bCN5obUrI1ahtwRCYcIHplAAjBkuk4eugkmEkL8ABo1ba7cr8CQAZlO50u11uEKQDxqT3kqH4gIAHMF3mhSd9fv8dECQVMwTNHFA4LJfkkAJrwnZ7NBiV4hDGoUBYhI4vFvZ5E-bkpAfWVKH5-IKUHRiBnTWZMJAbAAqNBiAGU7JsEXzPmJ9qDSsyKoa4ABVJKCHmIkbYPCEAhUTAOyhsHBsn5JOBwTDIHABpJkAiTLWOeYRQb1RroVmGgAUSYQy1Daw2AEp40zZn6YhAAMJsBjslPDdA4NjKTOvHPFm3gkDluAm+z1tMhXVwA3G00AKibLbbC1zKwLcH47ZLttm71HcAKSiGg4zgkz70zoywCma9ELBB7fbgAAIANSAjuM1eOHUbAAKbG8CW3qcc3lIJJMEzAByBo2EkVY-TkECFEKfQdAAXSfBN0C8GgB0cTNMzfOBP2-eDEIfMkLxRWVxx7Ksa3ZFDS0cLxMPQWJBAWJJM3Q2iXzQ6pfwbZJqLUbDmNY9iaBoQt+A4iiHWvTiuzUSRJEYkBsMBIQ2DITNMFxHDh3w9lCkBRDx3fABJAgqULe8b0BAgoGsXSPy-AyjJM8zLIvMyLzaG9pIrata0wOS7RABTlOEtg2IUyRgtmNQBl4wdIVrQTMwiqLFIk6LCz83tTVixxoB2TYd1fYcN2vXybynVsFBzPNVnWOArIIG9WyqrzOkQ6IkD6OA02KULoRQEU6PtGJFmEH4ag2ABHPZZFZS4fhycVqUDZajFKV10HXB0CiyEBdUcFAGW7e4IFiSLIlQQaoB2KAGEEaFZjmhbWUzfQ9orYtzAKdAmBLZjoV+QlChCEBENBe7BEe57fl6B1nUEO8czvECCBAu92OhBgEEoJA7y8O9KICmiAFJSeozB0cx7GccwPHKCYImYsLdHKBwHwQJqY70AYF0SxhuGXqwtRcfxpAqqvU1rLUSp2a0mTTXaX6jv+kBqmhh6ntF9AGaZphpeV+w5b6RWZfsVXeY1gYMm6kBkCSABrS5+iDf9kDITYzxgS5SvGvL+0O49xVAAPuxNhwzrZeyvB8S4PlAWIaDgQQLnQHjuz6D30Hm3E-W0v1SBmmHylkJ7BEOgZRpTtOM5ABLs9zkB8-ZOhoRoEulTLtAK8EKulAFnbk9T9PHGHmp3dmZwCR2TpNSG7TLi4lRIvHofBfQLBcHwIhSAoag6EYFh2F4ChJOGjZMAgAhxe0ghlHXtRiUBHR8ekOQaiupJkDD5uyyBggjIeQShYhwFxqGFk1MUibgXm8HIrt0De2sDAJAHttyLTYPHXwtdDQjy2LyWYkVcTe0mCAFOA8ADy7g1j9EuOacodJgRKkwJCZ25RQCMMcCQvEh0QxsA4QAdXUJoUa3DPhnXApBKA0FCT+Bzmufq90rCNB7sVf2f50BFRiOQyERdOG1C0SAQo+iXowgUDefQVjELkNZOyfqmi+LwUsfYjknJbGHTYbiCA8NDER0KFUBQ-RPEgihkAA)) is based on VL5 features (`params`/`expr`) and is still not yet passing VL5.1-schema completely. Therefor I don't think this can currently be done in Altair. But the future is bright🙏

**Reply by gregorhd** [reply]
Thanks both! That's encouraging. I'll keep watching this space :)
